src/components/track-changes/suggestion-list/Suggestion.tsx

```diff
@@ -34,53 +34,27 @@ export const Suggestion: React.FC<Props> = ({
   handleReset,
   handleClickSuggestion,
 }) => {
-  const [{ selectedSuggestion }] = useStore((store) => ({
-    user: store.user,
-    selectedSuggestion: store.selectedSuggestion,
+  const [{ selectedSuggestionID }, dispatch] = useStore((store) => ({
+    selectedSuggestionID: store.selectedSuggestionID,
   }))
 
-  const wrapperRef = useRef<HTMLLIElement>(null)
-
-  const [suggestionClicked, setSuggestionClicked] = useState(false)
-
-  const isSelectedSuggestion = useMemo(() => {
-    return !!(selectedSuggestion && selectedSuggestion === suggestion.id)
-  }, [selectedSuggestion, suggestion])
-
+  const wrapperRef = useRef(null)
   const [trackModalVisible, setModalVisible] = useState(false)
 
-  useEffect(() => {
-    if (isSelectedSuggestion && wrapperRef.current) {
-      setModalVisible(true)
-      if (!suggestionClicked) {
-        const wrapperRefElement = wrapperRef.current
-        wrapperRefElement.scrollIntoView({
-          behavior: 'auto',
-          block: 'start',
-          inline: 'start',
-        })
-      }
-    }
-    setSuggestionClicked(false)
-  }, [isSelectedSuggestion]) // eslint-disable-line react-hooks/exhaustive-deps
+  const isSelected = suggestion.id === selectedSuggestionID
+
+  const handleClick = (e: React.MouseEvent) => {
+    e.preventDefault()
+    dispatch({
+      selectedSuggestionID: suggestion.id,
+    })
+    setModalVisible(true)
+    handleClickSuggestion(suggestion)
```

Comment:
Would this just set the selected suggestion ID? But L48 already does that.

Reply:
No longer relevant.